.github/DESIGN.md

```diff
@@ -15,6 +26,32 @@ The user understands the basic mechanics of a CLI:
 - CTRL + C quits any running task
 - how to navigate and manipulate the filesystem via the command line: cd and mkdir
 
+## Components of a well designed command
+When creating a new command or subcommand there are a few things to keep in mind.
+
+### Reduce user input
+When trying to create parity with an existing GUI based feature, consider the larger task at hand and build your command to accomplish that goal. 
+
+For example when creating a `dev-store` the data model requires up to 8 fields to be filled in. When creating a CLI command to create a `dev-store`, don't ask the user for all 8 fields - auto-fill all 8 fields with smart generated data. If a field requires an address dont generate an animal name, try to get as close as possible to the format the data expects.
+
+### Allow arguments for key options
+Argumemnts allow the user to override a commands default execution. For example if we run `shopify populate products` the default will be to create 5 products. However if the user wanted more then 5 products to be generated they could use an argument to override the defaults which would look it would look like this `shopify populate products --count=20`.
```

Comment:
Typo fixes below, also isn't the `populate` default 10 products? Also, is the override `--count=20` or `--count 20`, with no `=`?

> Arguments allow the user to override a command’s default execution. For example if we run `shopify populate products`, the default will be to create 5 products. However, if the user wanted more than 5 products to be generated, they could use an argument to override the defaults: `shopify populate products --count=20`.

Reply:
Populate default is 5 in the CLI now.

Both `--count=20` or `--count 20` work, and I've seen the `=` used on other CLI help pages. but I'm easy with whatever our preferred way to represent it in the docs/help/etc pages. @tylerball @tanema @katiedavis